modules/33acrossAnalyticsAdapter.js

```diff
@@ -0,0 +1,475 @@
+  }
+
+  #flushTransactions() {
+    this.#clearSendTimeout();
+    this.#transactionsPending = new Set();
+    this.#onComplete();
+  }
+
+  // gulp-eslint is using eslint 6, a version that doesn't support private method syntax
+  // eslint-disable-next-line no-dupe-class-members
+  #clearSendTimeout() {
+    return clearTimeout(this.#sendTimeoutId);
+  }
+
+  // eslint-disable-next-line no-dupe-class-members
+  #restartSendTimeout() {
+    this.#clearSendTimeout();
+
+    this.#sendTimeoutId = setTimeout(() => {
+      if (this.#sendTimeout !== 0) {
+        log.warn(`Timed out waiting for ad transactions to complete. Sending report.`);
+      }
+
+      this.#flushTransactions();
+    }, this.#sendTimeout);
+  }
+}
+
+/**
+ * initialized during `enableAnalytics`
+ */
+const locals = {
+  /** @type {Object<string, TransactionManager>} - one manager per auction */
+  transactionManagers: {},
+  /** @type {AnalyticsCache=} */
+  cache: undefined,
+  /** sets all locals to undefined */
+  reset() {
+    this.transactionManagers = {};
+    this.cache = undefined;
+  }
+}
+
+/**
+ * @typedef {Object} AnalyticsAdapter
+ * @property {function} track
+ * @property {function} enableAnalytics
+ * @property {function} disableAnalytics
+ * @property {function} [originEnableAnalytics]
+ * @property {function} [originDisableAnalytics]
+ * @property {function} [_oldEnable]
+ */
+
+/**
+ * @type {AnalyticsAdapter}
+ */
+const analyticsAdapter = Object.assign(
+  buildAdapter({ analyticsType: 'endpoint' }),
+  { track: analyticEventHandler }
+);
+
+analyticsAdapter.originEnableAnalytics = analyticsAdapter.enableAnalytics;
+analyticsAdapter.enableAnalytics = enableAnalyticsWrapper;
+
+/**
+ * @typedef {Object} AnalyticsConfig
+ * @property {string} provider - set by pbjs at module registration time
+ * @property {Object} options
+ * @property {string} options.pid - Publisher/Partner ID
+ * @property {string} [options.endpoint=DEFAULT_ENDPOINT] - Endpoint to send analytics data
+ * @property {number} [options.timeout=DEFAULT_TRANSACTION_TIMEOUT] - Timeout for sending analytics data
+ */
+
+/**
+ * @param {AnalyticsConfig} config Analytics module configuration
+ */
+function enableAnalyticsWrapper(config) {
+  const { options } = config;
+
+  const pid = options.pid;
+  if (!pid) {
+    log.error('No partnerId provided for "options.pid". No analytics will be sent.');
+
+    return;
+  }
+
+  const endpoint = calculateEndpoint(options.endpoint);
+  this.getUrl = () => endpoint;
+
+  const timeout = calculateTransactionTimeout(options.timeout);
+  this.getTimeout = () => timeout;
+
+  locals.cache = {
+    pid,
+    auctions: {},
+  };
+
+  analyticsAdapter.originEnableAnalytics(config);
+}
+
+/**
+ * @param {string} [endpoint]
+ * @returns {string}
+ */
+function calculateEndpoint(endpoint) {
+  if (typeof endpoint === 'string' && endpoint.length > 0 && endpoint.startsWith('http')) {
```

Comment:
The `endpoint.length > 0` check is redundant, given the `endpoint.startsWith('http')` check.